### ISSUE TITLE
DHT20::read() Need Time out

### ISSUE THREAD

**Issue by 1220951120** [RobTillaart/DHT20#12]
```cpp
int DHT20::read()
{
  //  do not read to fast == more than once per second.
  uint32_t temp ;
  if (millis() - _lastRead < 1000)
  {
    return DHT20_ERROR_LASTREAD;
  }

  int status = requestData();
  if (status < 0) return status;
  //  wait for measurement ready
  temp = millis();
  while (isMeasuring())
  {
    if(millis()-temp >=1000)
    {
      Serial.print("DHT20 Time Out\n");
      return 0;
    }
    yield();
  }
  //  read the measurement
  status = readData();
  if (status < 0) return status;

  //  convert it to meaningful data
  return convert();
}
```

Thanks!

(updated code tags for syntax highlighting)

**Comment by RobTillaart** [comment]
Thanks for the issue.
What is the problem exactly?
What platform(s) did it occur?

**Comment by RobTillaart** [comment]
Note: Error code is already defined: **DHT20_ERROR_READ_TIMEOUT** in the DHT20.h file

But it should be not -14 but e.g. -300 as it should be clear it is an invalid temperature..

**Comment by RobTillaart** [comment]
Fixed in 0.2.3 release
